automap/src/comm_layer/mod.rs

```diff
@@ -251,36 +238,47 @@ mod tests {
                 AutomapErrorCause::ProtocolFailed,
             ),
             (
-                AutomapError::TemporaryMappingError(String::new()),
-                AutomapErrorCause::RouterFailure,
+                AutomapError::ProbeReceiveError(String::new()),
+                AutomapErrorCause::ProbeFailed,
+            ),
+            (
+                AutomapError::ProbeRequestError(AutomapErrorCause::ProbeFailed, String::new()),
+                AutomapErrorCause::ProbeFailed,
             ),
             (
                 AutomapError::ProbeServerConnectError(String::new()),
                 AutomapErrorCause::ProbeServerIssue,
             ),
             (
-                AutomapError::ProbeRequestError(AutomapErrorCause::ProbeFailed, String::new()),
-                AutomapErrorCause::ProbeFailed,
+                AutomapError::ProtocolError(String::new()),
+                AutomapErrorCause::ProtocolNotImplemented,
             ),
             (
-                AutomapError::ProbeReceiveError(String::new()),
-                AutomapErrorCause::ProbeFailed,
+                AutomapError::SocketBindingError(
+                    String::new(),
+                    SocketAddr::from_str("1.2.3.4:1234").unwrap(),
+                ),
+                AutomapErrorCause::UserError,
```

Comment:
I looked through the code. At the moment, it seems like the main reason we get `SocketBindingError`s is that there's already a Node running, and we can't open a socket because one is already open.

However, it seems as though that problem should have been solved by bringing in the `socket2` library. I'll look at it.

Reply:
It looks like with `socket2` in, the only thing that could reasonably cause a `SocketBindingError` would be some sort of mistake by the developers; so I switched this to `AutomapErrorCause::Unknown` and removed `AutomapErrorCause::UserError`.